manifests/config/postgres.pp

```diff
@@ -0,0 +1,29 @@
+# == Define: xylem::config::postgres
+#
+# [*postgres_host*]
+#   Host to create databases on.
+#
+# [*postgres_user*]
+#   User to create databases with.
+#
+# [*postgres_password*]
+#   Optional password for postgres user.
+#
+# [*postgres_secret*]
+#   Secret key for storing generated credentials.
+#
+class xylem::config::postgres(
+  $postgres_host     = undef,
+  $postgres_user     = undef,
+  $postgres_password = undef,
+  $postgres_secret   = undef,
```

Comment:
Because there aren't any sensible defaults, I think. This is all just the existing stuff moved around, I haven't actually changed any semantics.